ghostscript: update to 10.01.1.

mhmdanas opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **briefly**

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

tornaria commented:
Care to briefly explain the changed flags?

I've built and installed this locally, will report back in a week.

mhmdanas commented:
All of the flags I removed don't exist anymore, except for `--enable-dynamic`, which is deprecated (to be honest I removed this one and then kind of forgot to see what effects it had, if any, so it deserves some degree of scrutiny and testing).

tornaria commented:
I have this installed since a few weeks ago, I didn't test specifically but I didn't have any trouble.

Until now, since by the tiff update I had to rebuild. It'd be nice to get this merged.
